tools/jdbc/src/main/java/org/duckdb/DuckDBResultSet.java

```diff
@@ -115,6 +112,28 @@ private void check(int columnIndex) throws SQLException {
 
 	}
 
+	// Export the result set as an ArrowReader
+	public Object arrowExportStream(Object arrow_buffer_allocator, long arrow_batch_size) throws Exception {
```

Comment:
This shouldn't be declared as throwing exception, it's a pretty messy interface for the calling consumer. Ideally all the reflection exceptions should be handled internally and wrapped up in a `RuntimeException`, and the rest should be wrapped in `SQLException` per usual

Reply:
Great idea, updated PR accordingly